api/v1alpha3/azurecluster_default.go

```diff
@@ -95,6 +95,9 @@ func (c *AzureCluster) setSubnetDefaults() {
 	if nodeSubnet.RouteTable.Name == "" {
 		nodeSubnet.RouteTable.Name = generateRouteTableName(c.ObjectMeta.Name)
 	}
+	if nodeSubnet.NatGateway.Name == "" {
+		nodeSubnet.NatGateway.Name = generateNatGatewayName(c.ObjectMeta.Name)
```

Comment:
This mean we would be creating it by default. I'm not sure that's a behavior change we want, @devigned @alexeldeib thoughts?

Reply:
this is different behavior than described in the issue, ie. `Controller would check the new field and use it when not empty. It would be backward compatible.`

Reply:
i'd prefer the latter behavior, at least in the first iteration